lib/logstash/inputs/jdbc.rb

```diff
@@ -232,4 +249,23 @@ def update_state_file
     end
   end
 
+  private
+  # make sure the encoding is uniform over fields
+  def convert(key, value)
+    return value unless should_convert(key, value)
+    converter = find_converter_for value
+    converter.convert(value)
+  end
+
+  def should_convert(key, value)
+    value.is_a?(String) && (@columns_to_encode.empty? || @columns_to_encode.include?(key))
+  end
+
+  def find_converter_for(value)
+    unless @converters.keys.include?(value.encoding)
```

Comment:
ah, ok, I see, we rely on the actual String value encoding attribute to create a new converter for this encoding. I am not sure this is a reliable strategy?